internal/services/pam/pam.go

```diff
@@ -4,12 +4,15 @@ package pam
 import (
 	"context"
 	"encoding/json"
+	"errors"
 	"fmt"
+	"os/user"
 
 	"github.com/ubuntu/authd"
 	"github.com/ubuntu/authd/internal/brokers"
 	"github.com/ubuntu/authd/internal/log"
 	"github.com/ubuntu/authd/internal/users"
+	"github.com/ubuntu/authd/internal/users/cache"
```

Comment:
This is exposing the `users/cache` package (we previously fought so hard not to leak it). Shouldn't we remap the error to an exposed one in the `users` package and thus avoid leaking the cache one?